src/main/kotlin/com/exactpro/th2/codec/configuration/Configuration.kt

```diff
@@ -13,79 +13,53 @@
 
 package com.exactpro.th2.codec.configuration
 
+import com.exactpro.th2.codec.api.IPipelineCodecSettings
+import com.exactpro.th2.codec.api.IPipelineCodecSettingsTypeProvider
+import com.exactpro.th2.codec.util.load
 import com.exactpro.th2.common.schema.factory.CommonFactory
-import com.fasterxml.jackson.core.JsonParseException
-import com.fasterxml.jackson.core.type.TypeReference
 import com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES
-import com.fasterxml.jackson.databind.JsonMappingException
 import com.fasterxml.jackson.databind.ObjectMapper
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
 import com.fasterxml.jackson.module.kotlin.KotlinModule
-import org.apache.commons.lang3.StringUtils
-import java.io.IOException
-import java.nio.file.Files
-import java.nio.file.Files.newInputStream
-import java.nio.file.Paths
+import java.io.File
 
 internal val OBJECT_MAPPER: ObjectMapper = ObjectMapper(YAMLFactory()).apply { registerModule(KotlinModule()) }
     .configure(FAIL_ON_UNKNOWN_PROPERTIES, false)
 
-enum class ProcessorType { CUMULATIVE, SEQUENTIAL }
+class Configuration {
+    var codecSettings: IPipelineCodecSettings? = null
 
-class Configuration() {
-    var codecClassName: String? = null
+    val decoderInputAttribute: String = "decoder_in"
+    val decoderOutputAttribute: String = "decoder_out"
 
-    var codecParameters: Map<String, String>? = null
+    val encoderInputAttribute: String = "encoder_in"
+    val encoderOutputAttribute: String = "encoder_out"
 
-    var decoderInputAttribute: String = "decoder_in"
-    var decoderOutputAttribute: String = "decoder_out"
+    val generalDecoderInputAttribute: String = "general_decoder_in"
+    val generalDecoderOutputAttribute: String = "general_decoder_out"
 
-    var encoderInputAttribute: String = "encoder_in"
-    var encoderOutputAttribute: String = "encoder_out"
-
-    var generalDecoderInputAttribute: String = "general_decoder_in"
-    var generalDecoderOutputAttribute: String = "general_decoder_out"
-
-    var generalEncoderInputAttribute: String = "general_encoder_in"
-    var generalEncoderOutputAttribute: String = "general_encoder_out"
-
-    var decodeProcessorType: ProcessorType = ProcessorType.CUMULATIVE
+    val generalEncoderInputAttribute: String = "general_encoder_in"
+    val generalEncoderOutputAttribute: String = "general_encoder_out"
```

Comment:
I meant that they should be constant in my opinion. The user should not be able to change them

Reply:
Why a user shouldn't be allowed to change them?

Reply:
Because those attributes are used in configuration and description for pins that the box has. Changing them cause inconsistency between different codecs. If the user changes the attributes' names we will have to check what are the attributes' names for each codec before describing the pins.
In my opinion, this ability is redundant.

@Nikita-Smirnov-Exactpro what do you think?

Reply:
I agree with @OptimumOpium  and think we should make a common pin interface for codecs now.

Reply:
@Nikita-Smirnov-Exactpro  What do you mean by `common pin interface`?

Reply:
I think he meant that those pins should be the same for all codec and the user should not be able to change that

Reply:
I agree with @OptimumOpium